app/services/DynamoBanditData.scala

```diff
@@ -0,0 +1,119 @@
+package services
+
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.{Decoder, Encoder}
+import models.DynamoErrors.{DynamoError, DynamoGetError}
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, QueryRequest}
+import utils.Circe.dynamoMapToJson
+import io.circe.generic.auto._
+import zio.blocking.effectBlocking
+import zio.{ZEnv, ZIO}
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * Models for data received from DynamoDb
+ */
+
+case class VariantSample(variantName: String, annualisedValueInGBP: Double, annualisedValueInGBPPerView: Double, views: Double)
+case class TestSample(testName: String, variants: List[VariantSample], timestamp: String)
+object TestSample {
+  implicit val decoder = Decoder[TestSample]
+  implicit val encoder = Encoder[TestSample]
+}
+
+/**
+ * Models for data returned to the client
+ */
+
+// models the mean and views for each variant up to a certain timestamp
+case class EnrichedVariantSampleData(variantName: String, views: Double, mean: Double)
+case class EnrichedTestSampleData(timestamp: String, variants: List[EnrichedVariantSampleData])
+
+// Final mean and views for a variant
+case class VariantSummary(variantName: String, mean: Double, views: Double)
+
+case class BanditData(variantSummaries: List[VariantSummary], samples: List[EnrichedTestSampleData])
+object BanditData {
+  implicit val decoder = Decoder[BanditData]
+  implicit val encoder = Encoder[BanditData]
+}
+
+class DynamoBanditData(stage: String, client: DynamoDbClient) extends StrictLogging {
+  private val tableName = s"support-bandit-PROD"
```

Comment:
What happens on our CODE environment? Do we display PROD data?

Reply:
good spot, I was using PROD data for testing but I'll make this stage-aware